File: src/main/scala/jk_5/nailed/server/console/LoggerOutputStream.scala
package jk_5.nailed.server.console

import java.io.ByteArrayOutputStream

import org.apache.logging.log4j.{Level, Logger}

import scala.util.Properties

/**
 * No description given
 *
 * @author jk-5
 */
class LoggerOutputStream(val logger: Logger, val level: Level) extends ByteArrayOutputStream {

  val separator = Properties.lineSeparator

  override def flush(){
    this.synchronized{
      super.flush()
      val record = this.toString
      super.reset()
      if(record.length() > 0 && record != separator){
        logger.log(level, record)
      }
    }
  }
}

File: src/main/java/jk_5/nailed/server/logging/LoggerOutputStream.java
package jk_5.nailed.server.logging;

import org.apache.logging.log4j.Level;
import org.apache.logging.log4j.Logger;

import java.io.ByteArrayOutputStream;
import java.io.IOException;

public class LoggerOutputStream extends ByteArrayOutputStream {

    private static final String separator = System.getProperty("line.separator");
    private final Logger logger;
    private final Level level;

    public LoggerOutputStream(Logger logger, Level level) {
        this.logger = logger;
        this.level = level;
    }

    @Override
    public synchronized void flush() throws IOException {
        super.flush();
        String record = this.toString();
        super.reset();
        if(record.length() > 0 && !record.equals(separator)){
            logger.log(level, record);
        }
    }
}
